Bottom: b2ed403f32b6b21bd5be08c34f5c6afacbc0a428
Top:    5e0bce154175d684d1c327de4eddfbe6299dbde0
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-07-23 18:55:37 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 72a1660..ee3a19a 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -307,6 +307,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
